pkg/controller/util.go

```diff
@@ -326,3 +328,311 @@ func addToMap(m1, m2 map[string]string) map[string]string {
 	}
 	return m1
 }
+
+// returns the CloneRequest string which contains the pvc name (and namespace) from which we want to clone the image.
+func getCloneRequestPVC(pvc *v1.PersistentVolumeClaim) (string, error) {
+	cr, found := pvc.Annotations[AnnCloneRequest]
+	if !found || cr == "" {
+		verb := "empty"
+		if !found {
+			verb = "missing"
+		}
+		return cr, errors.Errorf("annotation %q in pvc \"%s/%s\" is %s\n", AnnCloneRequest, pvc.Namespace, pvc.Name, verb)
+	}
+	return cr, nil
+}
+
+func CreateCloneSourcePod(client kubernetes.Interface, image string, verbose string, pullPolicy string, cr string , pvc *v1.PersistentVolumeClaim) (*v1.Pod, error) {
+	strArr := strings.Split(cr, "/")
+    var ns string = ""
+	if strArr == nil || len(strArr) < 2 {
+		glog.V(Vdebug).Infof("Wrong CloneRequest Annotation")
+		return nil, nil
+	}
+	ns = strArr[0]
+	pvcName := strArr[1]
+	
+	pod := MakeCloneSourcePodSpec(image, verbose, pullPolicy, cr, pvcName)
+
+	pod, err := client.CoreV1().Pods(ns).Create(pod)
+	if err != nil {
+		return nil, errors.Wrap(err, "source pod API create errored")
+	}
+	glog.V(Vuser).Infof("cloning source pod \"%s/%s\" (image: %q) created\n", pod.Namespace, pod.Name, image)
+	return pod, nil
+}
+
+// return the clone source pod spec based on the target pvc.
+func MakeCloneSourcePodSpec(image, verbose, pullPolicy, cr, pvcName string) *v1.Pod {
+	// source pod name contains the pvc name
+	podName := fmt.Sprintf("%s-", CLONER_SOURCE_PODNAME)
+
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: podName,
+			Annotations: map[string]string{
+				AnnCloningCreatedBy: "yes",
+			},
+			Labels: map[string]string{
```

Comment:
_may_ want to add the "cdi.kubevirt.io" label added by David in pr [237](https://github.com/kubevirt/containerized-data-importer/pull/237)